adapter_darwin.go

```diff
@@ -0,0 +1,147 @@
+	a.cm.SetDelegate(a)
+	select {
+	case <-a.poweredChan:
+	case <-time.NewTimer(10 * time.Second).C:
+		return errors.New("timeout enabling CentralManager")
+	}
+	a.poweredChan = nil
+
+	// wait until powered?
+	a.pm.SetDelegate(a)
+
+	return nil
+}
+
+// CentralManager delegate functions
+
+// CentralManagerDidUpdateState when central manager state updated.
+func (a *Adapter) CentralManagerDidUpdateState(cmgr cbgo.CentralManager) {
+	// powered on?
+	if cmgr.State() == cbgo.ManagerStatePoweredOn {
+		close(a.poweredChan)
+	}
+
+	// TODO: handle other state changes.
+}
+
+// DidDiscoverPeripheral when peripheral is discovered.
+func (a *Adapter) DidDiscoverPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral,
+	advFields cbgo.AdvFields, rssi int) {
+	if a.peripheralFoundHandler != nil {
+		sr := makeScanResult(prph, advFields, rssi)
+		a.peripheralFoundHandler(a, sr)
+	}
+}
+
+// DidConnectPeripheral when peripheral is connected.
+func (a *Adapter) DidConnectPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral) {
+	// Make sure our delegate gets called for events related to this
+	// peripheral.
+	//prph.SetDelegate(d)
+
+	// Unblock now that we're connected.
+	a.connectChan <- prph
+}
+
+// DidDisconnectPeripheral when peripheral is disconnected.
+func (a *Adapter) DidDisconnectPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral, err error) {
+}
+
+// PeripheralManager delegate functions
+
+// PeripheralManagerDidUpdateState when state updated.
+func (a *Adapter) PeripheralManagerDidUpdateState(pmgr cbgo.PeripheralManager) {
+}
+
+// DidAddService when service added.
+func (a *Adapter) DidAddService(pmgr cbgo.PeripheralManager, svc cbgo.Service, err error) {
+}
+
+// DidStartAdvertising when advertising starts.
+func (a *Adapter) DidStartAdvertising(pmgr cbgo.PeripheralManager, err error) {
+}
+
+// DidReceiveReadRequest when read request received.
+func (a *Adapter) DidReceiveReadRequest(pmgr cbgo.PeripheralManager, cbreq cbgo.ATTRequest) {
+}
+
+// DidReceiveWriteRequests when write requests received.
+func (a *Adapter) DidReceiveWriteRequests(pmgr cbgo.PeripheralManager, cbreqs []cbgo.ATTRequest) {
+}
+
+// CentralDidSubscribe when central subscribed.
+func (a *Adapter) CentralDidSubscribe(pmgr cbgo.PeripheralManager, cent cbgo.Central, cbchr cbgo.Characteristic) {
+}
+
+// CentralDidUnsubscribe when central unsubscribed.
+func (a *Adapter) CentralDidUnsubscribe(pmgr cbgo.PeripheralManager, cent cbgo.Central, chr cbgo.Characteristic) {
+}
+
+// makeScanResult creates a ScanResult when peripheral is found.
+func makeScanResult(prph cbgo.Peripheral, advFields cbgo.AdvFields, rssi int) ScanResult {
+	var nu [16]byte
+	copy(nu[:], prph.Identifier())
+	uuid := NewUUID(nu)
+
+	var serviceUUIDs []UUID
+	for _, u := range advFields.ServiceUUIDs {
+		copy(nu[:], u)
+
+		parsedUUID := NewUUID(nu)
+		serviceUUIDs = append(serviceUUIDs, parsedUUID)
+	}
+
+	// It is never a random address. Is this correct?
```

Comment:
I have seen random addresses often in the wild, in particular for Nordic chips. But I guess that's irrelevant as it appears that CoreBluetooth doesn't expose MAC addresses.

Reply:
Changed to match how it is working on macOS.